Content.Server/Sports/PitchingMachineSystem.cs

```diff
@@ -0,0 +1,119 @@
+using System.Linq;
+using Content.Server.Popups;
+using Content.Server.Sports.Components;
+using Content.Server.Storage.Components;
+using Content.Server.Storage.EntitySystems;
+using Content.Shared.Throwing;
+using Content.Shared.Verbs;
+using Robust.Shared.Audio;
+using Robust.Shared.Physics;
+using Robust.Shared.Player;
+using Robust.Shared.Random;
+
+namespace Content.Server.Sports
+{
+    public sealed class PitchingMachineSystem : EntitySystem
+    {
+        [Dependency] private readonly PopupSystem _popupSystem = default!;
+        [Dependency] private readonly ThrowingSystem _throwingSystem = default!;
+        [Dependency] private readonly IRobustRandom _robustRandom = default!;
+        [Dependency] private readonly SharedAudioSystem _audioSystem = default!;
+        [Dependency] private readonly StorageSystem _storageSystem = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<PitchingMachineComponent, GetVerbsEvent<Verb>>(AddEjectVerb);
+            SubscribeLocalEvent<PitchingMachineComponent, GetVerbsEvent<AlternativeVerb>>(AddPowerVerb);
+        }
+
+        public override void Update(float frameTime)
+        {
+            base.Update(frameTime);
+            foreach (var ballLauncher in EntityQuery<PitchingMachineComponent>())
+            {
+                ballLauncher.AccumulatedFrametime += frameTime;
+
+                if (ballLauncher.AccumulatedFrametime < ballLauncher.CurrentLauncherCooldown)
+                    continue;
+
+                ballLauncher.AccumulatedFrametime -= ballLauncher.CurrentLauncherCooldown;
+                ballLauncher.CurrentLauncherCooldown = ballLauncher.ShootCooldown;
+
+                if (ballLauncher.IsOn)
+                    Fire(ballLauncher.Owner);
+            }
+        }
```

Comment:
This should just be using gun code.

Reply:
> This should just be using gun code.

I've made it to use as much guncode as i could, unfortunately I can't find a way to make the gun actually shoot itself.

Most stuff is the same except the Gun components firerate is used in the update loop and it's using BallisticAmmoProvider to hold the inserted items.

Shooting is just removing the items from the ammo container and doing TryThrow as I can't get the gun to shoot itself and gun projectiles are coded to just absolutely be yeeted out. Trythrow gives me more control (and its an actual throw)

Reply:
Gun code also uses trythrow.